homeassistant/components/asuswrt/diagnostics.py

```diff
@@ -0,0 +1,78 @@
+"""Diagnostics support for Asuswrt."""
+from __future__ import annotations
+
+from homeassistant.components.diagnostics import async_redact_data
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_PASSWORD, CONF_USERNAME
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers import device_registry as dr, entity_registry as er
+
+from .const import DATA_ASUSWRT, DOMAIN
+from .router import AsusWrtRouter
+
+TO_REDACT = {CONF_PASSWORD, CONF_USERNAME}
+
+
+async def async_get_config_entry_diagnostics(
+    hass: HomeAssistant, entry: ConfigEntry
+) -> dict:
```

Comment:
Could you make the return type more specific?
```suggestion
) -> dict[str, dict[str, Any]]:
```